services/provider/server/server.go

```diff
@@ -596,5 +526,119 @@ func (s *OCSProviderServer) RevokeStorageClassClaim(ctx context.Context, req *pb
 
 // GetStorageClassClaim RPC call to get the ceph resources for the StorageclassClaim.
 func (s *OCSProviderServer) GetStorageClassClaimConfig(ctx context.Context, req *pb.StorageClassClaimConfigRequest) (*pb.StorageClassClaimConfigResponse, error) {
-	return &pb.StorageClassClaimConfigResponse{}, nil
+	consumerClaimObj, err := s.storageClassClaimManager.Get(ctx, req.StorageConsumerUUID, req.StorageClassClaimName)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "failed to get storage class claim config %q for %q. %v", req.StorageClassClaimName, req.StorageConsumerUUID, err)
+		}
+		return nil, status.Errorf(codes.Internal, "failed to get storage class claim config %q for %q. %v", req.StorageClassClaimName, req.StorageConsumerUUID, err)
+	}
+
+	// TODO: add status.Phase validation here to check claim is in expected
+	// state or not before extracting the ceph resources from the StorageClassClaim CR status
+	conString, err := s.getExternalResourcesFromClaim(ctx, consumerClaimObj)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to get external resources. %v", err)
+	}
+
+	return &pb.StorageClassClaimConfigResponse{ExternalResource: conString}, nil
+}
+
+func (s *OCSProviderServer) getExternalResourcesFromClaim(ctx context.Context, storageClassClaim *ocsv1alpha1.StorageClassClaim) ([]*pb.ExternalResource, error) {
```

Comment:
Why is this a separate func? it is basically the implementation for `GetStorageClassClaimConfig` so I see no need to separate it. 

It has a single-use (single call site) and just complicates the reader's life by having him jump all over the file to find different parts of the implementation of the same action.

Can we fold this into the call site?

Reply:
it just increases the number of lines of code of a single function. thinking more about it even though it's a single time called function to provide abstraction from `Get` and extracting the information from the CR and increases the readability. if you still need this to be changed I can do it. let me know what do you think

Reply:
@Madhu-1 
Num of lines of codes per function should never ever be a consideration. 
There is a wide misconception that small functions are better! Flow is the important matric.

I am providing here two main points to back up the above claim:
1. By breaking stuff into functions you are creating a contract, in the form of function signature and variable scope,  Which by definition is a limiting factor to any code change in the future. And on top of that, the decision to break the flow this way will discourage future contributors to change this decision in the future even in cases where it is necessary (It is just human nature) 
2. There is no obvious location to search for to find the implementation of the action, it is split. It will be split for the reader and it will be split on the call stack when debugging. This just adds unnecessary complexity. Now it is true that on a case by case this seems irrelevant, but by repeating this pattern (splitting functions because you want them to be small) again and again, all over the codebase, the complexity aggregates and becomes meaningfully.

To clarify, I am not against refactoring stuff into functions. But functions, by definition, are a tool for reusability and here I cannot see a case for reusability, not now and not in the future.